backend/src/main/scala/cromwell/backend/io/JobPaths.scala

```diff
@@ -30,12 +31,21 @@ trait JobPaths {
 
   def workflowPaths: WorkflowPaths
   def returnCodeFilename: String = "rc"
-  def stdoutFilename: String = "stdout"
-  def stderrFilename: String = "stderr"
+  def defaultStdoutFilename = "stdout"
+  def defaultStderrFilename = "stderr"
+
+  final def stdoutFilename: String = jobKey.node match {
+    case t: TaskCallNode => t.callable.stdoutRedirection.getOrElse(defaultStdoutFilename)
+    case _ => ??? // TODO: JobKey should be typesafe
+  }
+  final def stderrFilename: String = jobKey.node match {
+    case t: TaskCallNode => t.callable.stdoutRedirection.getOrElse(defaultStderrFilename)
+    case _ => ??? // TODO: JobKey should be typesafe
+  }
   def scriptFilename: String = "script"
   def dockerCidFilename: String = "docker_cid"
-  
-  def jobKey: JobKey
+
+  def jobKey: CallKey
```

Comment:
Can you make it a `BackendJobDescriptorKey` ?

Reply:
not to pile on but a variable rename would be nice too

Reply:
@Horneth will change, thanks!
@danbills I think I still like `jobKey` here, IMO the `JobKey` trait is more in need of a rename (eg to `GraphNodeKey` or similar)